src/staff_ghosts.c

```diff
@@ -212,261 +212,275 @@ void func_80005310(void) {
-                D_80162DBC[0] = -1;
-                D_80162DB8 = 0;
+                sPlayerInputs = (u32*) &D_802BFB80.arraySize8[0][D_80162DCC][3];
+                sPlayerInputs[0] = -1;
+                sPlayerInputIdx = 0;
                 D_80162DDC = 0;
                 func_80091EE4();
-                if (D_80162DD4 == 0) {
+                if (gPlayerGhostDisabled == 0) {
                     func_80004FF8();
                 }
-                if (D_80162DD6 == 0) {
+                if (gCourseGhostDisabled == 0) {
                     func_80004EF0();
                 }
             }
         }
     }
 }
 
-void func_8000546C(void) {
-    u32 temp_a0;
-    u32 temp_a1;
+void process_replay_inputs(void) {
+    u32 inputs;
+    u32 stickBytes;
     UNUSED u16 unk;
-    u16 temp_v1;
-    s16 phi_v1;
-    s16 phi_v0 = 0;
+    u16 buttons_temp;
+    s16 stickVal;
+    s16 buttons = 0;
 
-    if (D_80162DB0 >= 0x1000) {
+    if (sReplayDataIdx >= 0x1000) {
         gPlayerOne->type = PLAYER_CINEMATIC_MODE | PLAYER_START_SEQUENCE | PLAYER_CPU;
         return;
     }
 
-    temp_a0 = D_80162DB4[D_80162DB0];
-    temp_a1 = temp_a0 & 0xFF;
+    inputs = replayData[sReplayDataIdx];
+    stickBytes = inputs & 0xFF;
 
-    if (temp_a1 < 0x80U) {
-        phi_v1 = (s16) (temp_a1 & 0xFF);
+    if (stickBytes < 0x80U) {
+        stickVal = (s16) (stickBytes & 0xFF);
     } else {
-        phi_v1 = (s16) (temp_a1 | (~0xFF));
+        stickVal = (s16) (stickBytes | (~0xFF));
     }
 
-    temp_a1 = (u32) (temp_a0 & 0xFF00) >> 8;
-    gControllerEight->rawStickX = phi_v1;
+    stickBytes = (u32) (inputs & 0xFF00) >> 8;
+    gControllerEight->rawStickX = stickVal;
 
-    if (temp_a1 < 0x80U) {
-        phi_v1 = (s16) (temp_a1 & 0xFF);
+    if (stickBytes < 0x80U) {
+        stickVal = (s16) (stickBytes & 0xFF);
     } else {
-        phi_v1 = (s16) (temp_a1 | (~0xFF));
+        stickVal = (s16) (stickBytes | (~0xFF));
     }
-    gControllerEight->rawStickY = phi_v1;
-    if (temp_a0 & 0x80000000) {
-        phi_v0 |= A_BUTTON;
+    gControllerEight->rawStickY = stickVal;
+    if (inputs & 1 << 31) {
+        buttons |= A_BUTTON;
     }
-    if (temp_a0 & 0x40000000) {
-        phi_v0 |= B_BUTTON;
+    if (inputs & 1 << 30) {
+        buttons |= B_BUTTON;
     }
-    if (temp_a0 & 0x20000000) {
-        phi_v0 |= Z_TRIG;
+    if (inputs & 1 << 29) {
+        buttons |= Z_TRIG;
     }
-    if (temp_a0 & 0x10000000) {
-        phi_v0 |= R_TRIG;
+    if (inputs & 1 << 28) {
+        buttons |= R_TRIG;
     }
-    temp_v1 = gControllerEight->buttonPressed & 0x1F0F;
-    gControllerEight->buttonPressed = (phi_v0 & (phi_v0 ^ D_80162DA8)) | temp_v1;
-    temp_v1 = gControllerEight->buttonDepressed & 0x1F0F;
-    gControllerEight->buttonDepressed = (D_80162DA8 & (phi_v0 ^ D_80162DA8)) | temp_v1;
-    D_80162DA8 = phi_v0;
-    gControllerEight->button = phi_v0;
+    buttons_temp = gControllerEight->buttonPressed & 0x1F0F;
```

Comment:
![image](https://github.com/user-attachments/assets/07026039-ea4a-4234-b7ca-53bde7a12857)

This can be broken by clicking on the bits at the bottom until there's just one at a time. Then check os_cont.h
So the first bit is 1 which is 
```c
CONT_CARD_ON | CONT_CARD_PULL | CONT_ADDR_CRC_ER |  U_CBUTTONS | R_JPAD | L_JPAD | D_JPAD | U_JPAD | START_BUTTON
```
This will clear any bit that isn't in the above whitelist. If the bit is already unset, it remains unset.

So if the user is pressing the A_BUTTON, then the A_BUTTON will be cleared (turned off). If the user is pressing D_JPAD then it stays pressed. If the user is not pressing D_JPAD its state stays the same (not pressed).

So 0x1F0F can be replaced with the above whitelist.

Reply:
Sounds good!